tests/v2/validation/networking/connectivity/network_policy_test.go

```diff
@@ -0,0 +1,145 @@
+		ClusterID: cluster.ID,
+		Name:      pingPodProjectName,
+	}
+
+	createdProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(n.T(), err)
+	require.Equal(n.T(), createdProject.Name, pingPodProjectName)
+	n.project = createdProject
+
+	names := newNames()
+
+	namespaceName := names.random["namespaceName"]
+	daemonsetName := names.random["daemonsetName"]
+
+	n.T().Logf("Creating namespace with name [%v]", namespaceName)
+	n.namespace, err = namespaces.CreateNamespace(n.client, namespaceName, "{}", map[string]string{}, map[string]string{}, n.project)
+	require.NoError(n.T(), err)
+	assert.Equal(n.T(), n.namespace.Name, namespaceName)
+
+	n.steveClient, err = n.client.Steve.ProxyDownstream(n.project.ClusterID)
+	require.NoError(n.T(), err)
+
+	testContainerPodTemplate := newPodTemplateWithTestContainer()
+
+	n.T().Logf("Creating a daemonset with the test container with name [%v]", daemonsetName)
+	daemonsetTemplate := shepworkloads.NewDaemonSetTemplate(daemonsetName, n.namespace.Name, testContainerPodTemplate, true, nil)
+	createdDaemonSet, err := n.steveClient.SteveType(workloads.DaemonsetSteveType).Create(daemonsetTemplate)
+	require.NoError(n.T(), err)
+	assert.Equal(n.T(), createdDaemonSet.Name, daemonsetName)
+
+	n.T().Logf("Waiting for daemonset [%v] to have expected number of available replicas", daemonsetName)
+	err = charts.WatchAndWaitDaemonSets(n.client, n.project.ClusterID, n.namespace.Name, metav1.ListOptions{})
+	require.NoError(n.T(), err)
+}
+
+func (n *NetworkPolicyTestSuite) TestPingPodsFromCPNode() {
+
+	// Get downstream cluster wrangler context
+	wc, err := n.client.WranglerContext.DownStreamClusterWranglerContext(n.project.ClusterID)
+	require.NoError(n.T(), err)
+
+	pods, err := wc.Core.Pod().List(n.namespace.Name, metav1.ListOptions{})
+	assert.NoError(n.T(), err)
+	assert.NotEmpty(n.T(), pods)
+
+	_, stevecluster, err := clusters.GetProvisioningClusterByName(n.client, n.clusterName, provisioninginput.Namespace)
+
+	query, err := url.ParseQuery("labelSelector=node-role.kubernetes.io/" + nodeRole + "=true")
+	assert.NoError(n.T(), err)
+
+	nodeList, err := n.steveClient.SteveType("node").List(query)
+	assert.NoError(n.T(), err)
+	assert.NotEmpty(n.T(), nodeList, err)
+
+	for _, machine := range nodeList.Data {
+		sshUser, err := sshkeys.GetSSHUser(n.client, stevecluster)
+		assert.NoError(n.T(), err)
+		assert.NotEmpty(n.T(), sshUser, errors.New("sshUser does not exist"))
+
+		sshNode, err := sshkeys.GetSSHNodeFromMachine(n.client, sshUser, &machine)
+		assert.NoError(n.T(), err)
+
+		n.T().Logf("Running ping on [%v]", machine.Name)
+
+		for i := 0; i < len(pods.Items); i++ {
+			podIP := pods.Items[i].Status.PodIP
+			pingExecCmd := pingCmd + " " + podIP
+			excmdLog, err := sshNode.ExecuteCommand(pingExecCmd)
+			if err != nil && !errors.Is(err, &ssh.ExitMissingError{}) {
+				assert.NoError(n.T(), err)
+			}
+			n.T().Logf("Log of the ping command {%v}", excmdLog)
```

Comment:
not blocking but, just noticed this log is a little much when looking at the jenkins job since there are a few pods its checked against. 
```
network_policy_test.go:137: Log of the ping command {PING 10.42.119.3 (10.42.119.3) 56(84) bytes of data.
11:30:15          64 bytes from 10.42.119.3: icmp_seq=1 ttl=63 time=0.289 ms
11:30:15          
11:30:15          --- 10.42.119.3 ping statistics ---
11:30:15          1 packets transmitted, 1 received, 0% packet loss, time 0ms
11:30:15          rtt min/avg/max/mdev = 0.289/0.289/0.289/0.000 ms
11:30:15          }
```
but if its helpful info feel free to leave it in.